src/libcrun/seccomp.c

```diff
@@ -340,6 +344,101 @@ seccomp_action_supports_errno (const char *action)
          || strcmp (action, "SCMP_ACT_TRACE") == 0;
 }
 
+static int
+calculate_seccomp_checksum (libcrun_container_t *container, unsigned int seccomp_gen_options, seccomp_checksum_t out, libcrun_error_t *err)
+{
+#if HAVE_GCRYPT
+  runtime_spec_schema_config_linux_seccomp *seccomp;
+  gcry_error_t gcrypt_err;
+  unsigned char *res;
+  gcry_md_hd_t hd;
+  size_t i;
+
+#  define PROCESS_STRING(X)                      \
+    do                                           \
+      {                                          \
+        if (X)                                   \
+          {                                      \
+            gcry_md_write (hd, (X), strlen (X)); \
+          }                                      \
+    } while (0)
+#  define PROCESS_DATA(X)                     \
+    do                                        \
+      {                                       \
+        gcry_md_write (hd, &(X), sizeof (X)); \
+    } while (0)
+
+  out[0] = 0;
+
+  if (container == NULL || container->container_def == NULL || container->container_def->linux == NULL)
+    return 0;
+
+  seccomp = container->container_def->linux->seccomp;
+  if (seccomp == NULL)
+    return 0;
+
+  gcrypt_err = gcry_md_open (&hd, GCRY_MD_SHA256, 0);
+  if (gcrypt_err)
+    return crun_make_error (err, EINVAL, "internal libgcrypt error: %s", gcry_strerror (gcrypt_err));
+
+  PROCESS_STRING (PACKAGE_VERSION);
+
+#  ifdef HAVE_SECCOMP
+  {
+    const struct scmp_version *version = seccomp_version ();
+
+    PROCESS_DATA (version->major);
+    PROCESS_DATA (version->minor);
+    PROCESS_DATA (version->micro);
+  }
+#  endif
+
+  PROCESS_DATA (seccomp_gen_options);
+
+  PROCESS_STRING (seccomp->default_action);
+  for (i = 0; i < seccomp->flags_len; i++)
+    PROCESS_STRING (seccomp->flags[i]);
+  for (i = 0; i < seccomp->architectures_len; i++)
```

Comment:
I wonder if this is enough, or if we also want to hash the current architecture? Say we seccomp->archtetures is ["i386"]. Will this give the same result when run on a i386 arch and a x86_64 arch?

Reply:
agreed, better be safe.  Would hashing the result of `uname(2)` be enough?

Reply:
pushed a new version where we also hash uname(2)